gnat2goto/driver/goto_utils.adb

```diff
@@ -329,19 +330,31 @@ package body GOTO_Utils is
                           I_Type          => Pointer_Type);
    end Offset_Array_Data;
 
-   function Float_Size (Entity_Esize : Uint) return Integer
+   function To_Float_Format (Float_Type : Irep) return Float_Format
    is
-      Unknown_Float_Width : exception;
+      Float_Width : constant Integer := Get_Width (Float_Type);
+      Unsupported_Float_Width : exception;
    begin
-      if Entity_Esize = 32 then
-         --  23-bit mantissa, 8-bit exponent
-         return 23;
-      end if;
-      if Entity_Esize = 64 then
-         --  52-bit mantissa, 11-bit exponent
-         return 52;
-      end if;
-      raise Unknown_Float_Width;
+      case Float_Width is
+         when 32 => return IEEE_32_Bit;
+         when 64 => return IEEE_64_Bit;
+         when others =>
+            Raise_Exception (Unsupported_Float_Width'Identity,
+                             "Unsupported float width: " &
+                               Integer'Image (Float_Width));
+      end case;
+   end To_Float_Format;
+
+   function Float_Size (Float_Type : Irep) return Integer
```

Comment:
Shouldn't this be something like `Float_Mantissa_Size`?

Reply:
Done.